### PR TITLE
Allow the `--isolated` pip flag to be optionally unset

### PR DESCRIPTION
**Pull request by UebelAndre**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

In adopting a wider use of `pip_parse` I've run into an issue where the pip config file which defines the cert location for my index_url is not being parsed, leading to failures when attempting to download packages.

Issue Number: N/A

## What is the new behavior?

This PR introduces flags for optionally disabling the use of the [--isolated](https://pip.pypa.io/en/stable/cli/pip/#cmdoption-isolated) flag that's currently always passed when invoking pip commands. Users can set `isolated = False` on uses of [pip_parse](https://github.com/bazelbuild/rules_python/blob/304b85cdb7be888a8aa8dce80c1e49374abdac18/docs/pip.md#pip_parse) and [pip_install](https://github.com/bazelbuild/rules_python/blob/304b85cdb7be888a8aa8dce80c1e49374abdac18/docs/pip.md#pip_install). Additionally, the environment variable `RULES_PYTHON_PIP_ISOLATED` can be set to a falsey value (`"0"`, `"False"`) to accomplish the same. The environment variable is intended to allow for this kind of configuration in `.bazelrc` files where users could define `common --repo_env=RULES_PYTHON_PIP_ISOLATED` to allow for this behavior in specific cases (like on CI machines).

I'm not sure if making `--isolated` optional is the best approach, but I do want to be able to rely on the machine's pip config to determine how to download things. The first alternative I reached for was to use `extra_pip_args` to pass [--cert](https://pip.pypa.io/en/stable/cli/pip/#cmdoption-cert) but next ran into an issue where `--cert` was different on different machines. I'm happy to hear some other alternatives though.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by UebelAndre** [comment]
Another option for fixing this without a change to the repo would be https://github.com/bazelbuild/rules_python/issues/335 but for me this is undesirable since since it requires an infra change for something that seems like inconsistent behavior in pip(?)

File: python/pip_install/pip_repository.bzl [hunk 1]
```diff
@@ -36,6 +36,20 @@ def _parse_optional_attrs(rctx, args):
     Returns: Augmented args list.
     """
 
+    # Determine whether or not to pass the pip `--isloated` flag to the pip invocation
+    use_isolated = rctx.attr.isolated
+
+    # The environment variable will take precedence over the attribute
+    isolated_env = rctx.os.environ.get("RULES_PYTHON_PIP_ISOLATED", None)
+    if isolated_env != None:
+        if isolated_env.lower() in ["", "0", "false"]:
```
Comment (hrfuller) [inline python/pip_install/pip_repository.bzl:45]: The empty string doesn't make sense as a falsey value to me. Can we keep this to set values?
```suggestion
        if isolated_env.lower() in ("0", "false"):
```

Comment (UebelAndre) [reply python/pip_install/pip_repository.bzl]: Yeah, sounds good to me 😃

**Comment by UebelAndre** [comment]
@hrfuller Hey, anything more I need to do for this PR?

**Comment by UebelAndre** [comment]
@hrfuller Hey, sorry to ping again but it'd be really great if this were to be merged 🙏 😅

**Review by hrfuller — APPROVED** [review]

**Comment by hrfuller** [comment]
LGTM. Merging. Thanks @UebelAndre

**Pull request merged**